src/utils/getFilteredPeople.tsx

```diff
@@ -0,0 +1,42 @@
+import { Person } from '../types';
+
+type Filters = {
+  sex: string | null;
+  query: string;
+  centuries: string[];
+};
+
+export const getFilteredPeople = (
+  people: Person[],
+  { sex, query, centuries }: Filters,
+) => {
+  let filteredPeople = [...people];
+
+  if (sex) {
+    filteredPeople = filteredPeople.filter(person => person.sex === sex);
+  }
+
+  const normalizedQuery = query.trim().toLowerCase();
+
+  if (normalizedQuery) {
+    filteredPeople = filteredPeople.filter(person => {
+      const normalizedName = person.name.toLowerCase();
+      const normalizedMotherName = person.motherName?.toLowerCase();
+      const normalizedFatherName = person.fatherName?.toLowerCase();
+
+      return (
+        normalizedName.includes(normalizedQuery) ||
+        normalizedMotherName?.includes(normalizedQuery) ||
+        normalizedFatherName?.includes(normalizedQuery)
+      );
+    });
+  }
+
+  if (centuries.length !== 0) {
+    filteredPeople = filteredPeople.filter(person =>
+      centuries.includes(String(Math.ceil(person.born / 100))),
```

Comment:
The filtering logic for centuries assumes that the `born` year is always positive. If there is any chance of dealing with historical data where years might be negative (BC), consider adding a check or handling for such cases to ensure accurate filtering.